src/mlpack/tests/test_tools.hpp

```diff
@@ -35,4 +34,9 @@
 
 #endif
 
+// Require the approximation L to be within a relative error of E respect to the
+// actual value R.
+#define REQUIRE_RELATIVE_ERR( L, R, E ) \
+    BOOST_REQUIRE_LE( abs((R) - (L)), (E) * (R))
```

Comment:
It's up to you (or, at least, I don't have a strong opinion).  The only issue with adding `REQUIRE_RELATIVE_ERR` is that it should be clearly documented, because another developer might expect something a little different out of a relative error condition.  Personally, I might consider using `BOOST_REQUIRE_CLOSE` with tolerance epsilon for KNN and with tolerance epsilon(1 + epsilon) for KFN (I think I did those calculations right) in order to avoid adding a new macro, but like I said, either way is fine, as long as the new macro is well-documented.